### PR TITLE
add js-dom task

### PR DESCRIPTION
**Pull request by allykrupina**
### js-dom task

[Demo](https://allykrupina.github.io/) |
[Code base](https://github.com/allykrupina/allykrupina.github.io)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by AMashoshyna** [comment]
Great job!

Please consider the following improvements:
- naming: functions names should contain verbs. Exception: words like "handler", "action" etc. which make it clear that the referred item is an action. `initial_cats` is a rather misleading name.
- lines 38-39 and 49-50 are almost identical. Please abstract this logic into a function.
- manipulating DOM nodes is expensive in terms of performance. Try to reduce queries to DOM whenever possible. Example: you make query `document.querySelectorAll(".menu-item")` on every loop iteration. Remember, that the browser needs to traverse the whole DOM to find the elements. Modern browsers are optimized and fast, but still it is a relatively slow operation. Instead you can make the query, save result in a variable and work with it.
- adding event listeners in a loop is ineffective. Please use event delegation instead: add one listener to container element. Also using `element.addEventListener` is more flexible than `onclick`.

**Comment by allykrupina** [comment]
> Great job!
> 
> Please consider the following improvements:
> 
> * naming: functions names should contain verbs. Exception: words like "handler", "action" etc. which make it clear that the referred item is an action. `initial_cats` is a rather misleading name.
> * lines 38-39 and 49-50 are almost identical. Please abstract this logic into a function.
> * manipulating DOM nodes is expensive in terms of performance. Try to reduce queries to DOM whenever possible. Example: you make query `document.querySelectorAll(".menu-item")` on every loop iteration. Remember, that the browser needs to traverse the whole DOM to find the elements. Modern browsers are optimized and fast, but still it is a relatively slow operation. Instead you can make the query, save result in a variable and work with it.
> * adding event listeners in a loop is ineffective. Please use event delegation instead: add one listener to container element. Also using `element.addEventListener` is more flexible than `onclick`.

Hello Anastasiya! 
I considered the improvements so I changes the logic of my code and rewrote it. Also I added a function call with a parameter to the handler. I would like to ask You about the last paragraph of improvements - Why  adding event listeners in a loop is inefficient? If the function is called with a specific parameter..

File: submissions/allykrupina/js-dom/js/main.js [hunk 1]
```diff
@@ -31,22 +31,17 @@ let all_cats = [
...
 ]
 
-function initial_cats() {
-    let handler = function() {
-        let cat_arr = document.querySelectorAll(".menu-item"),
-            index = Array.prototype.indexOf.call(cat_arr, this);
-            cat_title.textContent = all_cats[index].title;
-            cat_img.setAttribute("src", all_cats[index].url);
+function showCats() {
+    drowOneCat = (index) => {
```
Comment (AMashoshyna) [inline submissions/allykrupina/js-dom/js/main.js]: `const` keyword is missing. Without it it's a global variable.

**Comment by AMashoshyna** [comment]
In this particular case with only six elements you would not see any difference, but event delegation is a useful technique to learn, because handling many events in one place may make the code clearer and more elegant. Also for cases when you have many elements (let's say few thousand cells in a table) adding event listeners to every single one would make a difference in terms of performance.
So I agree that your solution is fine for this case.

File: submissions/allykrupina/js-dom/js/main.js (hunk 1 above)
Comment (allykrupina) [reply submissions/allykrupina/js-dom/js/main.js]: Fixed it.Thank you!

**Comment by AMashoshyna** [comment]
LGTM

**Pull request merged**